R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,407 @@
+                                                    height = 320 * min((options$nReps %/% 50 + 1), 5))
+  
+  
+  if (options$fixAxisLower >= options$fixAxisUpper) {
+    treePlotContainer$setError(gettext("Make sure to specify a lower bound that is lower than the upper bound for the x-axis."))
+    return()
+  }
+  
+  meanCI <- jaspContainer[["computedConfidenceIntervals"]][["object"]]
+  meanCI$Repetition <- rev(meanCI$Repetition)
+  meanCItrimmed <- meanCI[1:options$dataPlotShowN, ]
+  
+  myTicks <- jaspGraphs::getPrettyAxisBreaks(1:options$nReps)
+  myTicks[1] <- 1
+
+  p <- ggplot2::ggplot(meanCItrimmed,      
+                       ggplot2::aes(x = mean,
+                                    y = Repetition)) +
+    ggplot2::geom_point(ggplot2::aes(color = colorCI), size = 6, shape = 18) +
+    ggplot2::geom_errorbarh(ggplot2::aes(xmin = lower,
+                                         xmax = upper, 
+                                         color = colorCI)) +
+    ggplot2::scale_color_manual(values = c("A" = "black", "B" = "red")) +
+    ggplot2::geom_vline(data = data.frame(options$mu),
+                        ggplot2::aes(xintercept = options$mu),
+                        linetype = "dashed") +
+    ggplot2::scale_y_continuous(limits = c(0, options$nReps+1), 
+                                breaks = jaspGraphs::getPrettyAxisBreaks(1:options$nReps),
+                                labels = rev(myTicks)) +
+    ggplot2::xlab(gettextf("Observed mean and %s%% CI", options$confidenceIntervalInterval * 100)) 
+
+  if (options$fixAxisTreePlot) {
+    p <- p + ggplot2::scale_x_continuous(limits = c(options$fixAxisLower, options$fixAxisUpper))
+  }
+  
+  p <- p +
+    jaspGraphs::geom_rangeframe() + # add lines on the x-axis and y-axis
+    jaspGraphs::themeJaspRaw()      # add the JASP theme
+  
+  if(isTryError(p))
+    treePlotContainer[["treePlot"]]$setError(.extractErrorMessage(p))
+  else
+    treePlotContainer[["treePlot"]]$plotObject <- p
+  
+  return()
+}
+
+.plotDatasets <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["containerRainCloudPlots"]]) || !options$dataPlot)
+    return()
+  
+  rainData <- data.frame(y = unlist(jaspContainer[["simulatedDatasets"]][["object"]]), 
+                         group = rep(1:options$nReps, each = options$n))
+  
+  jaspContainer[["containerRainCloudPlots"]] <- createJaspContainer(gettext("Data plots"), 
+                                                                    dependencies = c("dataPlotShowN", "dataPlot"))
+  
+  rainData <- jaspContainer[["simulatedDatasets"]][["object"]]
+  
+  nRows <- min((options$dataPlotShowN %/% 3) + 1, 3)
+  nCols <- ifelse(nRows == 1, options$dataPlotShowN, 3)
+  addDots <- ifelse(options$dataPlotShowN > 9, TRUE, FALSE)
+  
+  matrixPlot <- createJaspPlot(title = gettext("Samples"), width = 960/(3/nCols), 
+                               height = 800/(3/nRows))
+  
+  plotList <- vector(mode = "list", length = nRows * nCols)
```

Comment:
you could also do 
```suggestion
  plotList <- matrix(list(), nRows, nCols)
```
and then you don't have to do the shenanigans with `listCounter` below